cnd/src/swap_protocols/herc20.rs

```diff
@@ -0,0 +1,263 @@
+use crate::{
+    asset,
+    swap_protocols::{
+        rfc003::{Secret, SecretHash},
+        state, SwapId,
+    },
+};
+use blockchain_contracts::ethereum::rfc003::Erc20Htlc;
+use chrono::NaiveDateTime;
+use futures::{
+    future::{self, Either},
+    Stream,
+};
+use genawaiter::sync::{Co, Gen};
+use std::collections::{hash_map::Entry, HashMap};
+use tokio::sync::Mutex;
+
+mod connector;
```

Comment:
This module doesn't really define the connector though, maybe `connector_impls` is a better name?